src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -864,9 +864,9 @@ def forward(
         hidden_states = self.self_attn_layer_norm(hidden_states)
 
         # Self Attention
-        # decoder uni-directional self-attention cached key/values tuple is at positions 1,2
-        self_attn_past_key_value = past_key_value[:2] if past_key_value is not None else None
-        # add present self-attn cache to positions 1,2 of present_key_value tuple
+        # decoder uni-directional self-attention cached key/values states are at position 0
+        self_attn_past_key_value = past_key_value[0] if past_key_value is not None else None
```

Comment:
The difficulty comes here from the fact that we're dealing with two sets of past key-values per decoder layer: one from the self-attention, and one from the cross-attention. The current solution uses a separate cache for each.